### PR TITLE
Fix ColorUtil's hued and huef

### PR DESCRIPTION
**Pull request by ZtereoHYPE**
They sometimes returned a negative value (not anymore) and the javadocs were slightly off, saying that the value ranged from 0..1 instead of 0..360 as is standard for hue and as these methods returned

### CODE REVIEW

File: src/main/java/net/shadew/util/misc/ColorUtil.java [hunk 1]
```diff
@@ -1370,21 +1370,23 @@ public static float huef(int color) {
         float cmin = MathUtil.min(r, MathUtil.min(g, b));
 
         float delta = cmax - cmin;
-        if (delta == 0) return 0;
```
Comment (FoxSamu) [inline src/main/java/net/shadew/util/misc/ColorUtil.java:1373]: Why did you change this into `return hue` if `hue` is always 0? That's an unnecessary variable allocation :p

File: src/main/java/net/shadew/util/misc/ColorUtil.java [hunk 2]
```diff
@@ -1370,21 +1370,23 @@ public static float huef(int color) {
...
+        float hue = 0F;
```
Comment (FoxSamu) [inline src/main/java/net/shadew/util/misc/ColorUtil.java]: `= 0`, the F is redundant

File: src/main/java/net/shadew/util/misc/ColorUtil.java [hunk 3]
```diff
@@ -1370,21 +1370,23 @@ public static float huef(int color) {
...
 
-        if (cmax == r) {
-            return (g - b) / delta % 6 * 60;
-        }
+        if (delta == 0) return hue;
 
-        if (cmax == g) {
-            return ((b - r) / delta + 2) * 60;
+        if (cmax == r) {
+            hue = (g - b) / delta % 6 * 60;
```
Comment (FoxSamu) [inline src/main/java/net/shadew/util/misc/ColorUtil.java]: Actually, this is the only line where hue returns negative since `(g-b)/delta` gives a value between -1 and 1; and any value below 0 here should become 6 + that value (java's modulo operator is just fucked up). Do the negativity check here instead of in the end so you don't need the `hue` variable and so you can avoid unnecessary branching.

```
if (cmax == r && g > b) {
    return (g - b) / delta * 60;
}  else if (cmax == g) {
    return ((b - r) / delta + 2) * 60;
} else if (cmax == b) {
    return ((r - g) / delta + 4) * 60;
} else {
    return ((g - b) / delta + 6) * 60;
}
```

File: src/main/java/net/shadew/util/misc/ColorUtil.java [hunk 4]
```diff
@@ -1398,17 +1400,19 @@ public static double hued(int color) {
         double cmin = Math.min(r, Math.min(g, b));
```
Comment (FoxSamu) [inline src/main/java/net/shadew/util/misc/ColorUtil.java:1400]: Same story as with the `float` version

**Review by FoxSamu — CHANGES_REQUESTED** [review]
I am rather picky about how this code is written because of performance and stuff so sorry for criticising your changes like this; tryna change what I proposed

File: src/main/java/net/shadew/util/misc/ColorUtil.java (hunk 2 above)
Comment (ZtereoHYPE) [reply src/main/java/net/shadew/util/misc/ColorUtil.java]: The more you know

File: src/main/java/net/shadew/util/misc/ColorUtil.java (hunk 1 above)
Comment (ZtereoHYPE) [reply src/main/java/net/shadew/util/misc/ColorUtil.java:1373]: I just noticed by reading, probably because in the process of trying to understand what it did I basically rewrote it all and then slowly reverted back to the original structure and that might be a remainder from the mess i did.

**Comment by ZtereoHYPE** [comment]
Oh yeah I totally understand! I also wanted to find a way to be able to have early returns to be closer to what you originally but that involved some annoying small code duplication. I'm still gonna go with that way and if u dont like revert.
After all this is my first stab at this stuff

**Comment by ZtereoHYPE** [comment]
oh shit github mobile only showed half the comments

feel free to squash merge to remove all the fuckups

File: src/main/java/net/shadew/util/misc/ColorUtil.java [hunk 5]
```diff
@@ -1373,13 +1373,14 @@ public static float huef(int color) {
 
         if (delta == 0) return 0;
 
-        if (cmax == r) {
-            float hue = (g - b) / delta % 6 * 60;
-            return hue < 0 ? hue + 360 : hue;
+        if (cmax == r && g > b) {
```
Comment (FoxSamu) [inline src/main/java/net/shadew/util/misc/ColorUtil.java]: I realised it should be `g >= b` so `0xFF0000` has hue 0 and not 360

File: src/main/java/net/shadew/util/misc/ColorUtil.java [hunk 6]
```diff
@@ -1401,13 +1402,14 @@ public static double hued(int color) {
 
         if (delta == 0) return 0;
 
-        if (cmax == r) {
-            double hue = (g - b) / delta % 6 * 60;
-            return hue < 0 ? hue + 360 : hue;
+        if (cmax == r && g > b) {
```
Comment (FoxSamu) [inline src/main/java/net/shadew/util/misc/ColorUtil.java]: again, use `>=`

**Review by FoxSamu — CHANGES_REQUESTED** [review]
This should be the last thing

File: src/main/java/net/shadew/util/misc/ColorUtil.java (hunk 5 above)
Comment (ZtereoHYPE) [reply src/main/java/net/shadew/util/misc/ColorUtil.java]: Oh my god i had even noticed that in my tests 🤦

**Review by FoxSamu — APPROVED** [review]
PERFECT

**Comment by ZtereoHYPE** [comment]
First try
